components/common/Input/Input.tsx

```diff
@@ -4,36 +4,48 @@ import { ChangeEvent, useEffect, useState } from "react";
 import "./input.scss";
 
 type InputProps = {
-	type: "password" | "email";
+	type: "password" | "email" | "search";
 	isError: boolean;
 	label: string;
 	value: string;
+	placeholder: string;
 	onChange: (e: ChangeEvent<HTMLInputElement>) => void;
 };
 
-const Input = ({ type, isError, label, value, onChange }: InputProps) => {
+const Input = ({
+	type,
+	isError,
+	label,
+	value,
+	onChange,
+	placeholder,
+}: InputProps) => {
 	const [mounted, setMounted] = useState(false);
 
 	//Hydration error with lastPass
 	useEffect(() => setMounted(true), []);
 	if (!mounted) return null;
 
-
 	// <button type="{type}" classname="{`button" button${="" variant="==" "primary"="" ?="" "--primary"="" :="" "--secondary"="" }="" ${isdisabled="" &&="" "button--disabled"}`}="">
 	// 		{label}
 	// 	</button>
 
 	return (
 		<div className="input__wrapper">
-			<label className="label" htmlFor={type}>{label}</label>
+			<label className="label" htmlFor={type}>
+				{label}
+			</label>
 			<input
 				id={type}
 				required
 				name={type}
 				value={value}
 				onChange={onChange}
-				className={`input ${isError && "input--error"}`}
+				className={`input ${isError && "input--error"} ${
+					placeholder && "input__placeholder"
+				}`}
 				type={type}
+				{...(placeholder ? { placeholder } : {})}
```

Comment:
tu coś jest nie tak, masz typ `placeholder: string;` to po co destrukturyzacja? :D 

wystarczy `placeholder={placeholder}` i najwyżej dać typ opcjonalny `placeholder?: string;`